lib/services/credentials/credential.dao.js

```diff
@@ -145,20 +145,47 @@ dao.removeCredential = function (id, type) {
   return db.del(buildIdKey(type, id));
 };
 
+/*
+ * Remove all credentials
+ *
+ * @id {String}
+ */
 dao.removeAllCredentials = function (id) {
   const dbTransaction = db.multi();
   const credentialTypes = Object.keys(config.models.credentials.properties);
-  const awaitAllPromises = [];
-  credentialTypes.forEach((type) => {
+  const awaitAllPromises = credentialTypes.map(async type => {
```

Comment:
I know that might be annoying to you but Express Gateway is using Promises and not any async/await constructs. This because it used to support Node 6 at the beginning. Although it would make sense to change, I do not think this is the right time to migrate.

Reply:
Any chance you could ask Al about this? For this thing in particular, async/await really helps keep it clean.

I was debating on whether to include it, so I specifically looked at your package.json and docker file. Both say that the engine is `"node": ">= 8.0.0"`. Then technically speaking, current version of LTS node is 10.

I don't think you have to migrate all at once. I think adding it gradually where it makes sense is fine.